### PR TITLE
Fix the URL.child docstring

### PR DESCRIPTION
**Pull request by exarkun**
https://twistedmatrix.com/trac/ticket/9021

### CODE REVIEW

File: src/twisted/python/_url.py [hunk 1]
```diff
@@ -528,9 +528,10 @@ def child(self, *segments):
 
         For example::
 
-            >>> (URL.fromText(u"http://localhost/a/b?x=y")
-                 .child(u"c", u"d").asText())
-            u'http://localhost/a/b/c?x=y'
+            >>> u = URL.fromText(u"http://localhost/a/b?x=y")
+            >>> v = u.child(u"c", u"d")
+            >>> print(v.asText())
+            u'http://localhost/a/b/c/d?x=y'
```
Comment (Julian) [inline src/twisted/python/_url.py]: Uh. 2 dumb questions.

1) epydoc doesn't have doctests then I take it?
2) Is the policy that we use Py2 reprs then? I assume the answer must be yes I suppose.

Regardless, lgtm.

**Review by Julian — APPROVED** [review]

File: src/twisted/python/_url.py (hunk 1 above)
Comment (rodrigc) [reply src/twisted/python/_url.py]: @Julian what do you mean by doctests?  There is a builder which generates documentation based on epydoc and reports errors: https://buildbot.twistedmatrix.com/builders/documentation/builds/2533

Comment (Julian) [reply src/twisted/python/_url.py]: @rodrigc If this were sphinx (well, or just regular stdlib doctest but those are dark times) then this error would have (or could have) failed an executable doctest which would have shown that the output didn't match what was executed.

So I was just guessing (well, learning) that apparently then nothing actually executes the documentation examples?

Comment (exarkun) [reply src/twisted/python/_url.py]: Generally speaking, Twisted doesn't use doctests.

Comment (exarkun) [reply src/twisted/python/_url.py]: > Is the policy that we use Py2 reprs then? I assume the answer must be yes I suppose.

No idea.  This docstring is a bit weird.  Not many Twisted docstrings contain this kind of example.  Reading this, I notice that it's still wrong - print wouldn't show the repr().

**Pull request merged**